File: app/src/main/java/dev/ragnarok/fenrir/link/types/OwnerLink.java
package dev.ragnarok.fenrir.link.types;

import androidx.annotation.NonNull;

public class OwnerLink extends AbsLink {

    public final int ownerId;

    public OwnerLink(int id) {
        super(id > 0 ? PROFILE : GROUP);
        ownerId = id;
    }

    @NonNull
    @Override
    public String toString() {
        return "OwnerLink{" +
                "ownerId=" + ownerId +
                '}';
    }
}

File: app/src/main/java/dev/ragnarok/fenrir/link/types/OwnerLink.kt
package dev.ragnarok.fenrir.link.types

class OwnerLink(val ownerId: Int) : AbsLink(if (ownerId >= 0) PROFILE else GROUP) {
    override fun toString(): String {
        return "OwnerLink{" +
                "ownerId=" + ownerId +
                '}'
    }
}